playground/app/components/ModalExample.vue

```diff
@@ -1,15 +1,33 @@
 <script setup lang="ts">
-const modal = useModal()
+import ModalExample2 from './ModalExample2.vue'
 
 defineProps<{
   count: number
 }>()
+
+const modalexample2 = useModal({
+  component: ModalExample2,
+  attrs: {
+    count: 1
+  }
+})
+
+const onOpenNested = () => {
+  modalexample2.open()
+}
```

Comment:
The idea, now, is that `useModal` returns an instance of the modal, and all methods are applied for that instance only.

in this case:
```ts
const modalexample2 = useModal({
  component: ModalExample2,
  attrs: {
    count: 1
  }
})

const onOpenNested = () => {
  modalexample2.open({count: 2})
}
```

modal is initialized with `{ attrs:{ count 1} }`, but when it is opened, the attrs are overwritten to `{ attrs:{ count 1} }`

I'm thinking it is more practicel to make the `attrs` optional when creating the modal instance with `useModal`, so that modal initialization is more compact.

The only caveat doing so is that if the attrs are required, there will be console warnings if they are not provided when its opened.